ngrinder-controller/src/main/java/org/ngrinder/agent/service/AgentPackageService.java

```diff
@@ -349,7 +347,7 @@ private void addAgentConfToTar(TarArchiveOutputStream tarOutputStream, String ba
 	 * @return true if it's jar
 	 */
 	public boolean isJar(File libFile) {
-		return StringUtils.endsWith(libFile.getName(), ".jar");
+		return libFile.getName().matches("^.*\\.jar(!)?$");
```

Comment:
Whay do we need regex match here? endsWith is not enough?

Reply:
currently version using spring-boot-gradle-plugin bootwar:requiresUnpack.

so endsWith is enough to check.